src/util/sock.h

```diff
@@ -115,6 +115,22 @@ class Sock
                                          void* opt_val,
                                          socklen_t* opt_len) const;
 
+    /**
+     * setsockopt(2) wrapper. Equivalent to
+     * `setsockopt(this->Get(), level, opt_name, opt_val, opt_len)`. Code that uses this
+     * wrapper can be unit tested if this method is overridden by a mock Sock implementation.
+     */
+    [[nodiscard]] virtual int SetSockOpt(int level,
+                                         int opt_name,
+                                         const void* opt_val,
+                                         socklen_t opt_len) const;
```

Comment:
Ignoring, the line would then become too long, and `GetSockOpt()` is already like that, so leaving it for consistency with surrounding code. Actually clang-format formats it like this, given a line length limit:

```cpp
    [[nodiscard]] virtual int            
    SetSockOpt(int level, int opt_name, const void* opt_val, socklen_t opt_len) const;
```

Reply:
> given a line length limit:

There isn't a limit, for this project the line isn't particularly long, and it would be annoying for all the functions with several params to be formatted this way. That said, thanks for responding.